### PR TITLE
docs: optimize the directory structure and add service discovery docs

### PR DESCRIPTION
**Pull request by cqqqq777**
#### What type of PR is this?

docs
#### Check the PR title.

- [x] This PR title match the format: \<type\>(optional scope): \<description\>
- [x] The description of this PR title is user-oriented and clear enough for others to understand.

#### (Optional) Translate the PR title into Chinese.
优化目录结构，补充服务注册与发现文档

#### (Optional) More detailed description for this PR(en: English/zh: Chinese).

en:
zh(optional):

### CODE REVIEW

File: content/en/docs/kitex/Tutorials/service-governance/service_discovery/_index.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+---
+title: "Service Discovery"
+date: 2023-11-30
+weight: 1
+keywords: ["Kitex", "Service Discovery", "Resolver"]
+description: Kitex provides extensions for service registration and discovery, and already supports many popular registries.
+---
+
+Kitex has completed multiple service discovery component through the support of community developers. Of course, it also supports DNS resolution and Static IP direct access mode. A strong and complete community ecology has been established for users to choose flexibly according to their needs.
```
Comment (felix021) [inline content/en/docs/kitex/Tutorials/service-governance/service_discovery/_index.md:9]: `completed` -> `implemented`
`component` -> `components`
`through` -> `with the help from`
`Of course` -> `Besides`

英文翻译找 AI 帮忙润色一下吧，不会显得过于 Chinglish。

File: content/zh/docs/kitex/Tutorials/service-governance/service_discovery/service-comb.md [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+
+## 安装
+
+```go
+go get github.com/kitex-contrib/registry-servicecomb
+```
+
+## 服务注册
+
+### 创建 Registy
```
Comment (GuangmingLuo) [inline content/zh/docs/kitex/Tutorials/service-governance/service_discovery/service-comb.md:17]: "Registy" should be "Registry".

File: content/zh/docs/kitex/Tutorials/service-governance/service_discovery/_index.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+linkTitle: "服务发现"
+weight: 1
+date: 2023-11-30
+keywords: ["服务注册与发现", "nacos", "consul", "etcd", "eureka", "polaris", "servicecomb", "zookeeper", "DNS"]
+description: "Kitex 框架提供服务注册与发现的扩展，目前已经支持与业界主流注册中心对接。"
+---
+
+Kitex 已经通过社区开发者的支持，完成了多种服务发现模式，当然也支持 DNS 解析以及 Static IP 直连访问模式，建立起了强大且完备的社区生态，供用户按需灵活选用。
+
+比如 DNS Resolver, 适合使用 DNS 作为服务发现的场景， 常见的用于 [Kubernetes](https://kubernetes.io/) 集群。
```
Comment (GuangmingLuo) [inline content/zh/docs/kitex/Tutorials/service-governance/service_discovery/_index.md:12]: DNS Resosolver 后面跟中文逗号
常见的用于 -> 常用于

File: content/zh/docs/kitex/Tutorials/service-governance/service_discovery/consul.md [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+
+func main() {
+	r, err := consul.NewConsulResolver("127.0.0.1:8500")
+	if err != nil {
+		log.Fatal(err)
+	}
+	c := hello.MustNewClient("hello", client.WithResolver(r), client.WithRPCTimeout(time.Second*3))
+	ctx := context.Background()
+	for {
+		resp, err := c.Echo(ctx, &api.Request{Message: "Hello"})
```
[the remark this answers is not in the document]
Comment (cqqqq777) [reply content/zh/docs/kitex/Tutorials/service-governance/service_discovery/consul.md:261]: 我感觉这个应该在完整项目示例中来体现，这里的重点是函数的使用，其他代码示例同理

**Review by AsterDY — APPROVED** [review]

**Pull request merged**